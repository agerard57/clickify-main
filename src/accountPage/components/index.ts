export { AccountLayout } from "./accountLayout";
export { CompanyInformation } from "./companyInformation";
export { CurrentPlan } from "./currentPlan";
