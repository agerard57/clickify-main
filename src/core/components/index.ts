export { TextField } from "./inputs";
export { PriceBox } from "./PriceBox";
export { HighlightText } from "./HighlightText";
