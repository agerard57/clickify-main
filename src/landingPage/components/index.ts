export { LandingPage } from "./LandingPage";
export { AboutSection, ProductSection, WelcomeSection, PricingSection, ContactSection } from "./sections";
