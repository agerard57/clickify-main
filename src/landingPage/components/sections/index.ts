export { ProductSection } from "./ProductSection";
export { WelcomeSection } from "./WelcomeSection";
