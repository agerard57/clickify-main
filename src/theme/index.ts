export { theme } from "./themeBuilder";
export { ButtonVariants, TypographyVariants } from "./typings";
