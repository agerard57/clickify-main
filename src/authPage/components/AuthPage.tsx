import React, { FC } from "react";
import { useTranslation } from "react-i18next";
import { useLocation } from "react-router-dom";

import { TextField } from "@/core";
import { ButtonVariants, TypographyVariants } from "@/theme";
import { Button, Typography } from "@mui/material";

import { AuthPages } from "../typings";
import { AuthLayout } from "./layout";

export const AuthPage: FC = () => {
  const { t } = useTranslation("AuthPage");
  const location = useLocation();

  const queryParams = new URLSearchParams(location.search);
  const pageParam = queryParams.get("page") as AuthPages;

  const currentlySelected =
    pageParam === "login" ? AuthPages.LOGIN : pageParam === "signUp" ? AuthPages.SIGN_UP : AuthPages.SIGN_UP;

  return (
    <AuthLayout currentlySelected={currentlySelected}>
      <>
        <Typography variant={TypographyVariants.TITLE} style={{ paddingBottom: 20 }}>
          {t(`${currentlySelected}.title`)}
        </Typography>
        <TextField
          autoFocus
          inputPosition="top"
          placeholder={t("inputs.email.placeholder")}
          inputLabel={t("inputs.email.label")}
          required
        />
        <TextField
          inputPosition="top"
          placeholder={t("inputs.password.placeholder")}
          inputLabel={t("inputs.password.label")}
          required
        />
        <Button variant={ButtonVariants.PRIMARY}>{t(`${currentlySelected}.title`)}</Button>
      </>
    </AuthLayout>
  );
};
