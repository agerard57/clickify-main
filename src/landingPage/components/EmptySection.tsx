import React, { FC } from "react";
import { useTranslation } from "react-i18next";

import { TypographyVariants } from "@/theme";
import styled from "@emotion/styled";
import { faCircleDown } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { Typography, useTheme } from "@mui/material";

import { SectionSpec } from "../typings";

interface Props {
  sectionSpec: SectionSpec;
  onClick?: () => void;
}

export const EmptySection: FC<Props> = ({ sectionSpec, onClick }) => {
  const { t } = useTranslation("LandingPage");
  const theme = useTheme();

  const BottomWrapper = styled.div`
    bottom: 40px;
    position: absolute;
    display: flex;
    flex-direction: column;
  `;

  const Icon = styled(FontAwesomeIcon)`
    height: 25px;
    margin-top: 15px;
    color: ${theme.app.landingPage.icon};
  `;

  return (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      {sectionSpec.sectionContent}

      {sectionSpec.nextSectionText && (
        <BottomWrapper>
          <Typography variant={TypographyVariants.SENTENCE}>{t(`${sectionSpec.name}.more`)}</Typography>
          <Icon icon={faCircleDown} onClick={onClick} />
        </BottomWrapper>
      )}
    </div>
  );
};
