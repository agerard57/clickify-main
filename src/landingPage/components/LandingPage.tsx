import styled from "@emotion/styled";
import React, { FC } from "react";
import { useTranslation } from "react-i18next";
import { HighlightText } from "@/core";
import { TypographyVariants } from "@/theme";
import { IconButton, Typography, useTheme } from "@mui/material";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCircleDown } from "@fortawesome/free-solid-svg-icons";
import { LandingPageImage } from "../assets";

export const LandingPage: FC = () => {
  const { t } = useTranslation("LandingPage");
  const theme = useTheme();

  const LandingPageWrapper = styled.div`
    align-items: center;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: absolute;
    width: 100%;
    height: -webkit-fill-available;
  `;

  const WelcomeContent = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 40px;
  `;

  const Image = styled.img`
    height: 400px;
    margin-top: 50px;
  `;

  const BottomWrapper = styled.div`
    position: absolute;
    bottom: 40px;
    display: flex;
    flex-direction: column;
  `;

  const Icon = styled(FontAwesomeIcon)`
    height: 25px;
    margin-top: 15px;
    color: ${theme.app.landingPage.icon};
  `;

  return (
    <LandingPageWrapper>
      <WelcomeContent>
        <HighlightText textVariant={TypographyVariants.TITLE} text={t("welcome.title")} />
        <Image src={LandingPageImage} />
      </WelcomeContent>
      <BottomWrapper>
        <Typography variant={TypographyVariants.SUBTITLE}>{t("welcome.more")}</Typography>
        <Icon icon={faCircleDown} />
      </BottomWrapper>
    </LandingPageWrapper>
  );
};
