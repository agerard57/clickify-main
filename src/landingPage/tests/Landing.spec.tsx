import { theme } from "@/theme";
import { ThemeProvider } from "@mui/material";
import React from "react";
import renderer from "react-test-renderer";

import { WelcomeSection } from "../components";

describe("Landing page - Welcome section", () => {
  test("should render welcome section properly", () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={theme}>
          <WelcomeSection />
        </ThemeProvider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
