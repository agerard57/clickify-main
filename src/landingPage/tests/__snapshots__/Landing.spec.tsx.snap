// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page - Welcome section should render welcome section properly 1`] = `
[
  <span
    className="MuiTypography-root MuiTypography-title css-1svodp3-MuiTypography-root"
  >
    Be closer to your clients,
in just a few 
    <span
      className="MuiTypography-root MuiTypography-title css-15e1932-MuiTypography-root"
    >
      clicks
    </span>
    !
  </span>,
  <img
    className="css-gdctic"
    src=""
  />,
]
`;
